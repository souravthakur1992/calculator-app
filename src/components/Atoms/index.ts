import Div from './Div';
import Span from './Span';
import Input from './Input';
import Button from './Button';
import NumberInput from './NumberInput'
import DropDownInput from './DropDownInput'

export default {
    Div,
    Span,
    Input,
    Button,
    NumberInput,
    DropDownInput
}