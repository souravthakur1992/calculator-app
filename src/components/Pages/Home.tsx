import React from 'react';
import Atoms from 'components/Atoms';
import Template from 'components/Template';
import Organisms from 'components/Organisms';

function Home(){
    return(
        <Atoms.Div display='flex' flexDirection='column'>
            <Template.Header />
            <Atoms.Div display='flex' padding="32px" justifyContent='center' alignItems='center'>
                <Organisms.Home />
            </Atoms.Div>
        </Atoms.Div>
    )
}

export default Home;