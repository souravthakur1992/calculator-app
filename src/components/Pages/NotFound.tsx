import React from 'react';
import { useNavigate } from 'react-router-dom';
import Lottie from 'lottie-react';
import { notFoundAnimation } from 'assets/lotties';
import Atoms from 'components/Atoms';

function NotFound(){
    const navigate = useNavigate();
    return(
        <Atoms.Div display='flex' flexDirection='column' justifyContent='center' alignItems='center'>
            <Atoms.Div display="flex" justifyContent='center' width="100%">
                <Atoms.Span fontSize="30px" fontWeight="700">404 Error, Not Found</Atoms.Span>
            </Atoms.Div>
            <Atoms.Div width="600px" height="400px" marginTop="30px">
                <Lottie animationData={notFoundAnimation} />
            </Atoms.Div>
            <Atoms.Div display='flex' flexDirection='column' gap="30px" alignItems='center' width="100%">
                <Atoms.Span>The page you requested has disappeared or you have taken the wrong route.</Atoms.Span>
                <Atoms.Button onClick={() => {navigate('/')}} width="200px" height="40px" borderRadius="12px" border="2px solid black">Go To Homepage</Atoms.Button>
            </Atoms.Div>
        </Atoms.Div>
    )
}

export default NotFound;