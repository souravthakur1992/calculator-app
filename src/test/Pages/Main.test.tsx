import { render, screen } from '@testing-library/react';
import { Home } from 'components/Pages';
const renderer = require('react-test-renderer');
test('render Main Page', () => {
    render(<Home />);

    const h1Tag = screen.getByText(/CALCULATOR REACT APP/i);
    expect(h1Tag).toBeInTheDocument();
})
describe("Snapshot test for Home page", () => {
    it("Matches DOM Snapshot", () => {
        const domTree = renderer.create(<Home />).toJSON();
        expect(domTree).toMatchSnapshot();
    });
});