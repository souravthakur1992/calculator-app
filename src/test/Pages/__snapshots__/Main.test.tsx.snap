// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Home page Matches DOM Snapshot 1`] = `
<div
  className="sc-bgqQcB iJyUKE"
  display="flex"
>
  <div
    className="sc-bgqQcB gcCEJg"
    display="flex"
    height="76px"
    width="calc(100% - 24px)"
  >
    <div
      className="sc-bgqQcB glEEYE"
      display="flex"
    >
      <div
        className="sc-bgqQcB jegIso"
        display="flex"
        height="50px"
        width="50px"
      >
        <div />
      </div>
      <span
        className="sc-gTRrQi kNNORz"
        fontFamily="Montserrat"
        fontSize="24px"
        fontWeight="700"
      >
        CALCULATOR APP
      </span>
    </div>
  </div>
  <div
    className="sc-bgqQcB PJhlA"
    display="flex"
  >
    <div
      className="sc-bgqQcB iyWsvV"
    >
      <div
        className="sc-bgqQcB eCkknC"
        display="flex"
        height="100%"
        width="100%"
      >
        <div
          className="sc-bgqQcB bRiYAl"
          display="flex"
        >
          <span
            className="sc-gTRrQi ghqOhS"
            fontFamily="Montserrat"
            fontSize="32px"
            fontWeight="700"
          >
            CALCULATOR REACT APP
          </span>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="sc-bgqQcB cbMStY"
            display="flex"
            width="100%"
          >
            <div
              className="sc-bgqQcB bmduCb"
              display="flex"
              width="125px"
            >
              <input
                className="sc-dicizt fufbmn"
                name="num1"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyUp={[Function]}
                placeholder="Number 1"
                step="any"
                type="number"
                value=""
                width="125px"
              />
            </div>
            <div
              className="sc-bgqQcB bmduCb"
              display="flex"
              width="125px"
            >
              <input
                className="sc-dicizt fufbmn"
                name="num2"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyUp={[Function]}
                placeholder="Number 2"
                step="any"
                type="number"
                value=""
                width="125px"
              />
            </div>
            <div
              className="sc-bgqQcB gXpjif"
              display="flex"
              width="200px"
            >
              <select
                className="sc-iVCKna fBHkMa"
                name="operator"
                onBlur={[Function]}
                onChange={[Function]}
                value=""
                width="200px"
              >
                <option
                  value=""
                >
                  Select an operatorâ€¦
                </option>
                <option
                  value="+"
                >
                  ADD
                </option>
                <option
                  value="-"
                >
                  SUBTRACT
                </option>
                <option
                  value="*"
                >
                  MULTIPLY
                </option>
                <option
                  value="/"
                >
                  DIVIDE
                </option>
              </select>
            </div>
            <div
              className="sc-bgqQcB fbePWa"
              display="flex"
              width="250px"
            >
              <input
                className="sc-dicizt likbKM"
                disabled={true}
                fontFamily="monospace"
                fontSize="1rem"
                fontWeight="700"
                placeholder="Press Calculate For Answer"
                value=""
                width="250px"
              />
            </div>
          </div>
          <div
            className="sc-bgqQcB bAttSk"
            display="flex"
            width="100%"
          >
            <button
              className="sc-fFGjHI jncEbr"
              color="#ffffff"
              fontFamily="Roboto"
              fontSize="1rem"
              fontWeight="400"
              height="44px"
              type="submit"
              width="410px"
            >
              CALCULATE
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
